docs/adr/adr-0022-new-icon-system.md

```diff
@@ -0,0 +1,255 @@
+
+While the implementation is quite straightforward, it unfortunately comes with downsides.
+
+### As static assets with no backing icon library
+
+As the number of icons increases, the risk of disparateness increases as well, managing them becomes harder and harder.
+This solution does not scale well.
+
+And icon creation and/or picking can be pretty tedious.
+
+### As image tags
+
+When referenced using an `<img>` tag, it is not possible to update a SVG `stroke` and `fill` properties using CSS (more details in [this article about SVG customization](https://css-tricks.com/the-many-ways-to-change-an-svg-fill-on-hover-and-when-to-use-them/)).
+
+Thus, color customization is quite restricted and forces us to have a one file per color usage for each icon. Another consequence is making theme implementation more difficult (especially for a dark theme).
+
+### And more
+
+From a technical point, icons are imported with the `new URL('./foo.svg', import.meta.url).href` syntax in our components.
+
+This created more complexity in our stacks with a dedicated [Rollup](https://www.npmjs.com/package/@web/rollup-plugin-import-meta-assets) plugin in our components and a dedicated [Babel](https://www.npmjs.com/package/@babel/plugin-syntax-import-meta) plugin in some projects where our components are used.
+
+Same consideration on Vue.js projects where this syntax triggers errors: more details on this topic in [issue #130](https://github.com/CleverCloud/clever-components/issues/130).
+
+## The solution
+
+We decided to:
+
+- change how we displayed icons,
+- change how the assets were processed,
+- add a backing icon library.
+
+This was done by:
+
+- creating an icon component,
+- adding a rollup plugin,
+- adding a backing icon library.
+
+### The icon component
+
+We wanted a technical layer helping us with displaying icons.
+Naturally, it comes as a new (atomic) component, as part of our Clever Components.
+More precisely as a Web Component built on top of [Lit](https://lit.dev/).
+
+This `<cc-icon>` component allows us to have a better control on how the icon is displayed through:
+
+- component properties: on size and skeleton mode,
+- CSS custom properties: on size (as an override mechanism) and color.
+
+These two points are made possible because the component renders the SVG content inline.
+Having a dedicated component also helps to implement more accessible icons (see below).
+
+This SVG rendering point brings us to another part of the new icon system: the Rollup plugin.
+
+### The Rollup plugin
+
+We wanted to change the way icons were loaded in components, in order to get rid of the `new URL('./foo.svg', import.meta.url).href` syntax.
+The simplest and effective way in our case is by importing the icon as an ES object: that's where the plugin comes into play.
+
+Its roles are to:
+
+- parse the icon assets,
+- clean and optimize them (with [SVGO](https://github.com/svg/svgo)),
+- transform them into ES objects,
+- write them in a specific file.
```

Comment:
This list could mention (in the last item or in another one) that the produced file is a valid ESM (or ES module) file.